File: CodingSamples/Java/Platform/ThreadTest/Program2.java
class Program {

	private static String client;

	private static void handleJob(int id) {
		System.out.printf("Thread<%x> accepted job<%d> for %s%n", Thread.currentThread().hashCode(), id, client);
		Worker.doWork(id);
		System.out.printf("Thread<%x> finished job<%d> for %s%n", Thread.currentThread().hashCode(), id, client);
	}

	public static void main(String[] args) throws Exception {
		int n = args.length > 0 ? Integer.parseInt(args[0]) : 1;
		Thread child = new Thread(() -> {
			client = "Jack";
			handleJob(n);
		});
		child.start();
		client = "Jill";
		handleJob(2);
	}
}

File: CodingSamples/DotNet/Platform/ThreadTest/DemoApp/Program2.cs
namespace DemoApp;
class Program
{
    static string client;

    static void HandleJob(int id)
    {
        Console.WriteLine("Thread<{0}> has accepted job<{1}> for {2}", Thread.CurrentThread.ManagedThreadId, id, client);
        Worker.DoWork(id);
        Console.WriteLine("Thread<{0}> has finished job<{1}> for {2}", Thread.CurrentThread.ManagedThreadId, id, client);
    }

    static void Main(string[] args)
    {
        int n = args.Length > 0 ? int.Parse(args[0]) : 1;
        Thread child = new Thread(() => 
        {
            client = "Jack";
            HandleJob(n);            
        });
        child.Start();
        client = "Jill";
        HandleJob(2);
    }
}
